### PR TITLE
updateTx only needs to work in `IsRecentEra era`

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Change era constraint for `updateTx` from `IsShelleyBasedEra` to `IsRecentEra`

### Comments

- `updateTx` should also be moved to some module in the `Write.Tx` hierarchy eventually

### Issue Number

ADP-2849 but for `updateTx` instead of `signTx`.

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet/Shelley/Compatibility/Ledger.hs [hunk 1]
```diff
@@ -322,65 +319,32 @@ toMaryTxOut (TxOut addr bundle) =
...
-    -> Maybe (Hash "Datum")
     -> Babbage.BabbageTxOut StandardBabbage
-toBabbageTxOut (TxOut addr bundle) = \case
-    Nothing ->
-        Babbage.BabbageTxOut
-            (toLedger addr)
-            (toLedger bundle)
-            Babbage.NoDatum
-            Ledger.SNothing
-    Just (Hash bytes) ->
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Wallet/Shelley/Compatibility/Ledger.hs:353]: `Just` case never used

File: lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs [hunk 2]
```diff
@@ -873,113 +871,43 @@ updateTx (Cardano.Tx body existingKeyWits) extraContent = do
...
         , Babbage.txfee =
             modifyFee $ Babbage.txfee ledgerBody
         }
-    ShelleyBasedEraAlonzo -> ledgerBody
+    RecentEraAlonzo -> ledgerBody
         { Alonzo.outputs = Alonzo.outputs ledgerBody
             <> StrictSeq.fromList
-                ( Cardano.toShelleyTxOut era
-                . Cardano.toCtxUTxOTxOut
-                . toCardanoTxOut era <$> extraOutputs
```
Comment (Anviking) [inline lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:915]: Avoided going through `cardano-api` types here, and went for `Compatibility.Ledger.toAlonzoTxOut` directly.

Comment (Anviking) [reply lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs:915]: Btw, some day, we should be able to remove lots of things in the wallet's `Shelley.Compatibility` module.

**Review by Unisay — APPROVED** [review]
LGTM!

**Pull request merged**